src/pages/blog/neon.md

```diff
@@ -0,0 +1,497 @@
+```
+
+From here, we can make a `.env.development` file (which will be auto-ignored by the gitignore) and add the database URL to it:
+
+```
+USERS_DATABASE_URL=<neon url here>
+```
+
+From here, we can run `rails db:migrate` to setup the database with the ActiveRecord internal information and then `rails s` to start the development environment up.
+
+From here, you should have a working rails environment that we can develop with:
+
+![Rails hello world screen](/images/neon/hello_world_rails.png)
+
+## Making the User model and tieing the multi-DB architecture into it
+
+To setup the User model, we are going to want to setup Rails using the devise framework. This will allow us to easily be able to setup user authentication within our application. To do that, add the following to your `Gemfile` then run `bundle`:
+
+```rb
+# Add devise authentication
+gem 'devise', '~> 4.9.4'
+```
+
+From here, we need to run `rails g devise:install` to install devise specific files into our project.
+
+Now we have done that, lets generate a model for the User and attach authentication to it. We will give the database information as JSON to the record. This is where we will store the users database information:
+```
+➜  note-taking git:(main) rails g model User database_information:json
+      invoke  active_record
+      create    db/migrate/20241012124645_create_users.rb
+      create    app/models/user.rb
+      invoke    test_unit
+      create      test/models/user_test.rb
+      create      test/fixtures/users.yml
+➜  note-taking git:(main) rails g devise User
+      invoke  active_record
+      create    db/migrate/20241012124742_add_devise_to_users.rb
+      insert    app/models/user.rb
+       route  devise_for :users
+```
+
+Next, we will run `rails db:migrate` to store our changes and add `database_information` to `config/initializers/filter_parameter_logging.rb`:
+```rb
+# Be sure to restart your server when you modify this file.
+
+# Configure parameters to be partially matched (e.g. passw matches password) and filtered from the log file.
+# Use this to limit dissemination of sensitive information.
+# See the ActiveSupport::ParameterFilter documentation for supported notations and behaviors.
+Rails.application.config.filter_parameters += [
+  ...,
+  :database_information,
+]
+```
+
+So now we have our users, but what about our notes? Let's add logic to put each user's notes in a separate Postgres database.
+
+There are several advantages to  this database-per-tenant architecture:
+
+- **Security by Design:** There is significantly less risk of harm due to accidentally destroying/fetching other customers data. This is because instead of worrying about multiple columns when you do things to a users notes, you just have to think about one. Additionally, you don't have to worry about the risk of someone finding another users notes by decrementing their own note ID since the users notes are pulled from the signed in record.
```

Comment:
```suggestion
- **Security by Design:** There is less risk of harm due to accidentally destroying/fetching other customer's data. This is because instead of worrying about multiple columns when you do things to a user's notes, you just have to think about one. Additionally, you have less risk of someone finding another users notes _( for example, by decrementing their own note ID)_ since the user's notes are pulled from the signed in record.
```